### PR TITLE
Move some of HTTPHandler stuff to separate source files in order to reuse it in PrometheusRequestHandler

### PR DESCRIPTION
**Pull request by vitlibar**
### Changelog category:
- Not for changelog

### Changelog entry:
Move some of `HTTPHandler` stuff to separate source files in order to reuse it in `PrometheusRequestHandler`.
This PR is required for https://github.com/ClickHouse/ClickHouse/pull/64183

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit ecfe6fddcfbe68477fb10fabc4b0252e08d198ef with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/66067/ecfe6fddcfbe68477fb10fabc4b0252e08d198ef/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Style check</td><td>Runs a set of checks to keep the code style clean. If some of tests failed, see the related log from the report</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/66067/ecfe6fddcfbe68477fb10fabc4b0252e08d198ef/style_check.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: base/poco/Net/src/NameValueCollection.cpp [hunk 1]
```diff
@@ -102,9 +102,9 @@ const std::string& NameValueCollection::get(const std::string& name, const std::
 		return defaultValue;
 }
 
-const std::vector<std::reference_wrapper<const std::string>> NameValueCollection::getAll(const std::string& name) const
+const std::vector<std::string> NameValueCollection::getAll(const std::string& name) const
```
Comment (antonio2368) [inline base/poco/Net/src/NameValueCollection.cpp:105]: const for return value looks strange

File: src/Server/HTTP/authenticateUserByHTTP.cpp [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+{
+    extern const int AUTHENTICATION_FAILED;
+    extern const int BAD_ARGUMENTS;
+    extern const int SUPPORT_IS_DISABLED;
+}
+
+
+namespace
+{
+    String base64Decode(const String & encoded)
```
Comment (antonio2368) [inline src/Server/HTTP/authenticateUserByHTTP.cpp:37]: `Common/Base64.h`

File: src/Server/HTTP/authenticateUserByHTTP.cpp [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+    {
+        /// It is prohibited to mix different authorization schemes.
+        if (has_http_credentials)
+            throw Exception(ErrorCodes::AUTHENTICATION_FAILED,
+                            "Invalid authentication: it is not allowed "
+                            "to use SSL certificate authentication and Authorization HTTP header simultaneously");
+        if (has_credentials_in_query_params)
+            throw Exception(ErrorCodes::AUTHENTICATION_FAILED,
+                            "Invalid authentication: it is not allowed "
+                            "to use SSL certificate authentication and authentication via parameters simultaneously simultaneously");
```
Comment (antonio2368) [inline src/Server/HTTP/authenticateUserByHTTP.cpp:100]: in exception we mention SSL certificate but this triggers with any headers right not just SSL, right?

File: src/Server/HTTP/authenticateUserByHTTP.cpp [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+    else // I.e., now using user name and password strings ("Basic").
+    {
+        if (!current_credentials)
+            current_credentials = std::make_unique<BasicCredentials>();
+
+        auto * basic_credentials = dynamic_cast<BasicCredentials *>(current_credentials.get());
+        if (!basic_credentials)
+            throw Exception(ErrorCodes::AUTHENTICATION_FAILED, "Invalid authentication: expected 'Basic' HTTP Authorization scheme");
+
+        basic_credentials->setUserName(user);
```
Comment (antonio2368) [inline src/Server/HTTP/authenticateUserByHTTP.cpp:206]: `chassert(!user.empty());`

**Review by antonio2368 — APPROVED** [review]

File: base/poco/Net/src/NameValueCollection.cpp (hunk 1 above)
Comment (vitlibar) [reply base/poco/Net/src/NameValueCollection.cpp]: `const` removed

File: src/Server/HTTP/authenticateUserByHTTP.cpp (hunk 2 above)
Comment (vitlibar) [reply src/Server/HTTP/authenticateUserByHTTP.cpp]: Corrected, now `Common/Base64.h` is used

File: src/Server/HTTP/authenticateUserByHTTP.cpp (hunk 3 above)
Comment (vitlibar) [reply src/Server/HTTP/authenticateUserByHTTP.cpp]: The error messages were not correct, I've fixed them.

File: src/Server/HTTP/authenticateUserByHTTP.cpp (hunk 4 above)
Comment (vitlibar) [reply src/Server/HTTP/authenticateUserByHTTP.cpp:202]: Added

**Pull request merged**